### PR TITLE
[Hot compaction] Add upgrade logic for TsFileManagement

### PR DESCRIPTION
**Pull request by zhanglingzhe0820**

### CODE REVIEW

File: docs/zh/SystemDesign/StorageEngine/TsFileManagement.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	* 生成目标文件 {first_file_name}-{max_level_num - 1}.tsfile
+	* 生成合并日志 .hot_compaction.log
+	* 记录是全局合并
+	* 记录目标文件
+	* 进行合并（记录日志 device - offset）
+	* 记录完成合并
+* 循环判断每一层的文件是否要合并到下一层
+	* 生成目标文件 {first_file_name}-{level + 1}.tsfile
+	* 生成合并日志 .hot_compaction.log
+	* 记录是全局合并
```
Comment (qiaojialin) [inline docs/zh/SystemDesign/StorageEngine/TsFileManagement.md:68]: fix

File: docs/zh/SystemDesign/StorageEngine/TsFileManagement.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+* 如果日志文件不存在
+	* 无需恢复
+
+## LevelStrategy 例子
+
+设置 max_file_num_in_each_level = 3，tsfile_manage_strategy = LevelStrategy， max_level_num = 3，此时文件结构为，第0层、第1层、第2层，其中第2层是不再做热合并的稳定的文件列表
+
+### 完全根据 level 合并的情况
+假设此时整个系统中有5个文件，最后一个文件没有关闭,则其结构及顺序分布如下
+0: t2-0 t3-0 t4-0
```
Comment (qiaojialin) [inline docs/zh/SystemDesign/StorageEngine/TsFileManagement.md:97]:
```suggestion
level-0: t2-0 t3-0 t4-0
```
```suggestion
0: t2-0 t3-0 t4-0
```

**Review by qiaojialin — APPROVED** [review]

File: docs/zh/SystemDesign/StorageEngine/TsFileManagement.md (hunk 1 above)
Comment (zhanglingzhe0820) [reply docs/zh/SystemDesign/StorageEngine/TsFileManagement.md:68]: I fix it to level

File: docs/zh/SystemDesign/StorageEngine/TsFileManagement.md (hunk 2 above)
Comment (zhanglingzhe0820) [reply docs/zh/SystemDesign/StorageEngine/TsFileManagement.md:97]: okay, I will add it

**Pull request merged**